packages/lib/src/ref/core.ts

```diff
@@ -154,35 +182,61 @@ function getBackRefs(target: object, refType?: RefConstructor<object>): Observab
  *
  * @typeparam T Referenced object type.
  * @param target Node the references point to.
- * @param [refType] Pass it to filter by only references of a given type, or do not to get references of any type.
+ * @param refType Pass it to filter by only references of a given type, or do not to get references of any type.
+ * @param options Options.
  * @returns An observable set with all reference objects that point to the given object.
  */
 export function getRefsResolvingTo<T extends object>(
   target: T,
-  refType?: RefConstructor<T>
+  refType?: RefConstructor<T>,
+  options?: {
+    updateAllReferencesIfNeeded?: boolean
+  }
 ): ObservableSet<Ref<T>> {
   assertTweakedObject(target, "target")
 
   const refTypeObject = refType as RefConstructor<object> | undefined
+
+  if (options?.updateAllReferencesIfNeeded && isReactionDelayed()) {
+    // in this case the reference update might have been delayed
+    // so we will make a best effort to update them
+    allRefs.forEach((ref) => ref.forceUpdateBackRefs())
```

Comment:
> I try to avoid it because marking a node as a root store is (in theory) optional, but very practical to be able to get that hook working. [...]
>
> Source: https://github.com/xaviergonz/mobx-keystone/issues/233#issuecomment-825216171

Or instead of maintaining the `allRefs` set via the `onAttachedToRootStore` hook, what about [your other suggestion using `walkTree`](https://github.com/xaviergonz/mobx-keystone/issues/233#issuecomment-823530686)?

```ts
walkTree(fastGetRoot(target), n => {
  if (n instanceof Ref) {
    n.forceUpdateBackRefs()
  }
)
```

It probably is more expensive to walk the tree instead of iterating over just ref nodes, but it would work for trees that are not root stores. Backrefs across trees would still not work though (not my use case at the moment, just saying). Could the performance be improved by using [`computedWalkTreeAggregate`](https://github.com/xaviergonz/mobx-keystone/blob/0444b0f95bb316d0e3446e2b03d41d6f226ae5e4/packages/lib/src/parent/walkTree.ts#L138-L161) due to caching?

Reply:
Walk tree would be slower. The computed one would be not as slow, but still slower too.

Reply:
I'd say it's your call. I always register a root store, but the (computed) tree walk would be more consistent with the rest of the library regarding keeping root stores optional.

Reply:
Even if you call a tree walk, where would you start the walk from?
- From the root of the target model? (the ref might be in another tree)
- From the root of all root stores? (same result as if it used onAttachedToRootStore but slower)

Maybe if the target has a root (but is not part of a root store) use walk tree of the target tree + all root stores to update?
Dunno, I feel it is over complicating for a rare use case of an already rare use case :)

Reply:
Ah, I forgot that the solution that uses `onAttachedToRootStore` is able to resolve references even across multiple trees.

Looks good then, let's keep it as it is now. :slightly_smiling_face: